### DISCUSSION TITLE
string-indexed objects in message templates not rendered on seq server

### DISCUSSION THREAD

**Discussion by iainsproat** [datalust/seq-tickets#2258] [Help] [answered]
I'd like to use a message template with a string-indexed object e.g. `{req['method']} {req['path']} request {requestStatus} in {responseTime} ms`.

The metadata included with this example message is of the following structure:
```
{
  req: {
    method: 'POST',
    path: '/graphql'
  },
  requestStatus: 'completed',
  responseTime: 23
}
```

The seq documentation suggests that string-indexed objects should work in templates, but doesn't provide any examples: https://docs.datalust.co/docs/template-syntax#basic-syntax (aside note, the further reading at the bottom of this documentation page are broken links)

After searching datalust's repositories, I found [some examples of string-indexed objects](https://github.com/datalust/seq-app-mail/blob/98d85b28c06cb02002f67d5b786f36c55679c333/test/Seq.Syntax.Tests/Cases/expression-evaluation-cases.asv#L27) though I'm not certain if this is the same syntax as used in the server.

See example of rendered message when running locally using [pino-pretty](https://github.com/pinojs/pino-pretty) on node, and what I was hoping would appear on seq server:
<img width="999" alt="Screenshot 2024-08-23 at 10 47 07" src="https://github.com/user-attachments/assets/3e574c4f-c468-4b8c-b61f-92f81d3441ea">

Unfortunately, the string-indexed objects within the template are not rendered correctly on seq server (version `2024.3.1`)
<img width="618" alt="Screenshot 2024-08-23 at 10 59 13" src="https://github.com/user-attachments/assets/69bbfcb0-8bcd-40d0-9cd3-3b16d7064b65">

Should I expect seq server to support string-indexed objects in templates?
If so, what is the correct syntax?

**Comment by nblumhardt** [comment, marked as answer]
Hi Iain, thanks for your message.

The "Template syntax" page describes the syntax for generating emails or HTTP requests in output apps like the SMTP email and HTTP request plug-ins.

The syntax supported in log messages is https://messagetemplates.org, which doesn't include support for indexers etc.

I'll check over the docs and try to make this clearer - sorry about the confusion!

Best regards,
Nick

**Reply by iainsproat** [reply]
ok, thank you. The features of the log syntax are clearer now.
I landed on the template syntax page after a Google search for 'seq template', so wasn't aware of the context it applied to - I see now it's in the 'sending notifications' category in the sidebar.
I've since found the references to the log syntax: https://www.google.com/search?q=site%3Adatalust.co+%22messagetemplates.org%22
